src/components/StageForm.js

```diff
@@ -12,25 +12,38 @@ class StageForm extends Component {
       props.getStore(),
       Object.keys(props.form)
     );
-
     this.onChange = this.onChange.bind(this);
     this.isValidated = this.isValidated.bind(this);
   }
 
   setupState(store, keys) {
     let newState = {};
 
-    // Create "error" keys for each form value, and set it to null
+    // Create "error" keys for each form value
     keys.forEach((key) => {
-      let errorName = key + '_error';
-
-      newState[key] = store[key];
-      newState[errorName] = null;
+      newState[key] = store[key] === undefined ? "" : store[key];
     });
 
     return newState;
   }
 
+  componentDidUpdate(prevProps) {
+    // If step has changeds
+    if (prevProps.step !== this.props.step) {
```

Comment:
I believe this is the only place step is used... probably you could get by with just comparing prevProps.form with this.props.form and remove step.